controllers/common.go

```diff
@@ -0,0 +1,117 @@
+package controllers
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/aws/aws-sdk-go/service/ecr/ecriface"
+	"github.com/quintilesims/d.ims.io/models"
+	"github.com/zpatrick/fireball"
+)
+
+func listRepositories(e ecriface.ECRAPI) ([]string, error) {
+	repositories := []string{}
+
+	fn := func(output *ecr.DescribeRepositoriesOutput, lastPage bool) bool {
+		for _, repository := range output.Repositories {
+			repositories = append(repositories, aws.StringValue(repository.RepositoryName))
+		}
+
+		return !lastPage
+	}
+
+	if err := e.DescribeRepositoriesPages(&ecr.DescribeRepositoriesInput{}, fn); err != nil {
+		return nil, err
+	}
+
+	return repositories, nil
+}
+
+func getRepositoryPolicy(e ecriface.ECRAPI, repositoryName string) (*models.PolicyDocument, error) {
+	policyDoc := &models.PolicyDocument{}
+
+	getPolicyInput := &ecr.GetRepositoryPolicyInput{}
+	getPolicyInput.SetRepositoryName(repositoryName)
+	if err := getPolicyInput.Validate(); err != nil {
+		return nil, err
+	}
+
+	getPolicyOutput, err := e.GetRepositoryPolicy(getPolicyInput)
+	if err != nil {
```

Comment:
This seems a little confusing - the inverse logic with a comment may be more readable:
```go
// continue execution if the policy does not exist
output, err := e.GetRepositoryPolicy(input)
if err != nil {
    if awsErr, ok := err.(awserr.Error); !ok || awsErr.Code() != ecr.ErrCodeRepositoryPolicyNotFoundException {
        return nil, err
    }
}
```

Reply:
✅